### PR TITLE
chore(deps): update to `age` version `0.11`

### PR DESCRIPTION
**Pull request by JarvisCraft**
# Description

This updates to the latest `age` version `0.11` which reduces allocations.

This also updates `flake.lock` to resolve the recent breakage.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -63,13 +63,13 @@ dependencies = [
...
-version = "0.10.0"
+version = "0.11.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "edeef7d7b199195a2d7d7a8155d2d04aee736e60c5c7bdd7097d115369a8817d"
+checksum = "2020562e68d7a02c2743707b262c62484b340a296924a5e4146d5a0a96ca8103"
 dependencies = [
  "aes",
  "aes-gcm",
- "age-core",
+ "age-core 0.11.0",
```
Comment (CertainLach) [inline Cargo.lock]: Why is there two versions of age-core now?

Comment (JarvisCraft) [reply Cargo.lock]: Fixed

File: crates/fleet-base/src/fleetdata.rs [hunk 2]
```diff
@@ -73,16 +72,14 @@ pub struct FleetSharedSecret {
 }
 
 /// Returns None if recipients.is_empty()
-pub fn encrypt_secret_data(
-	recipients: impl IntoIterator<Item = impl Recipient + Send + 'static>,
+pub fn encrypt_secret_data<'a>(
+	recipients: impl IntoIterator<Item = &'a (impl Recipient + Send + 'static)>,
```
Comment (CertainLach) [inline crates/fleet-base/src/fleetdata.rs]: Looks fishy, can you simplify this associated type?

Comment (CertainLach) [reply crates/fleet-base/src/fleetdata.rs]: Whatever with_recipients accepts should be fine I think.

Comment (JarvisCraft) [reply crates/fleet-base/src/fleetdata.rs]: Simplified it to `Item = &'a dyn Recipient` and changed the callers to reflect the change

**Review by CertainLach — APPROVED** [review]

**Pull request merged**